pkg/controller/accountclaim/accountclaim_controller.go

```diff
@@ -99,6 +103,30 @@ func (r *ReconcileAccountClaim) Reconcile(request reconcile.Request) (reconcile.
 		return reconcile.Result{}, err
 	}
 
+	// Add finalizer to the CR in case it's not present (e.g. old accounts)
+	if !contains(accountClaim.GetFinalizers(), accountClaimFinalizer) {
+		err := r.addFinalizer(reqLogger, accountClaim)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+	}
+
+	// Check if accountClaim is being deleted, this will trigger the account reuse workflow
+	if accountClaim.DeletionTimestamp != nil {
```

Comment:
The problem with returning after the cleanup, and removing the finalizer in another run of the reconcile loop is you lock the accountClaim deletion process until you remove the finalizer. That would lock UHC waiting to delete the claim.

It would be 10-20 minutes until you're able to queue again to remove the finalizer, specially as you have other accounts being created/deleted.